File: quicksort.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
int Partition(int A[],int l,int r)
{
    int X,Y,Pivot,temp;
    X=l;
    Y=r;
    Pivot=A[l];
    while(X<Y)
    {
        while (A[X]<=Pivot && X<=r)
        {
            X++;
        }
        while(A[Y]>Pivot && Y>l)
        {
            Y--;
        }
        if(X<Y)
        {
            temp=A[X];
            A[X]=A[Y];
            A[Y]=temp;
            
        }
    }
    temp=A[l];
    A[l]=A[Y];
    A[Y]=temp;
    return Y;
}
void quickSort(int A[],int l,int r)
{
    int p;
    if(l<r)
    {
        p=Partition(A,l,r);
        quickSort(A,l,p-1);
        quickSort(A,p+1,r);
    }
}

void display(int A[],int n)
{
    int i;
    for(i=0;i<n;i++)
    {
        printf("%d ",A[i]);
    }
    printf("\n");
}
int main()
{
    int A[100000],n,i;
    clock_t start,end,diff;
    printf("Enter n:");
    scanf("%d",&n);
    for(i=0;i<n;i++)
    {
        A[i]=rand();
    }
    display(A,n);
    start =clock();
    quickSort(A,0,n-1);
    end=clock();
    display(A,n);
    diff=end-start;
    printf("Time Taken is %f second\n",(float)diff/CLOCKS_PER_SEC);
    return 0;
}

File: quicksort.py
import time
import random

def partition(A, l, r):
    x = l
    y = r
    pivot = A[l]
    while x < y:
        while A[x] <= pivot and x <= r:
            x += 1
        while A[y] > pivot and y > l:
            y -= 1
        if x < y:
            temp = A[x]
            A[x] = A[y]
            A[y] = temp
    temp = A[l]
    A[l] = A[y]
    A[y] = temp
    return y

def quickSort(A, l, r):
    if l < r:
        p = partition(A, l, r)
        quickSort(A, l, p - 1)
        quickSort(A, p + 1, r)

def display(A, n):
    for i in range(0, n):
        print(A[i], end=" ")
    print()

A = [0] * 100000
print("Enter n:", end=" ")
n = int(input())
for i in range(0, n):
    A[i] = random.randint(0, 10)
display(A, n)
start = time.time()
quickSort(A, 0, n - 1)
end = time.time()
display(A, n)
diff = end - start
print(f"Time Taken is {diff:.6f} second")
